Neos.Neos/Classes/FrontendRouting/DimensionResolution/Resolver/AutoUriPathResolverFactory.php

```diff
@@ -56,22 +56,25 @@ public function create(
                     . ' For everything more advanced, please manually configure UriPathResolver in Settings.yaml.'
             );
         }
+        if (count($contentDimensions) === 0) {
```

Comment:
just a minor minor thing, but we already count the dimensions in line 53 so I would suggest to introduce a new variable like `$numberOfContentDimensions = count($contentDimensions);`

Also I wonder if we should turn this into an early return along the lines of
```php
if ($numberOfContentDimensions === 0) {
  return UriPathResolver::createForNoDimensions();
}
```
}

Reply:
I think we could even introduce a match clause because of the beauty of it

Reply:
this is the way

Reply:
okay, it's maybe a bit much for a match, but switch looks good enough

Reply:
yes i agree ;) @bwaidelich lets merge?